file.go

```diff
@@ -133,3 +135,44 @@ func TouchFile(f *os.File) error {
 	}
 	return nil
 }
+
+func GetFinalPathName(fileName string, openFlags uint32, finalPathFlags uint32) (result string, err error) {
+	isXP, _ := IsWindowsXP()
+	if isXP {
+		// Todo: resolve symlink target on Windows XP
+		return fileName, nil
+	}
+	file, e := wrappers.CreateFile(
+		syscall.StringToUTF16Ptr(fileName),
+		wrappers.GENERIC_READ,
+		wrappers.FILE_SHARE_READ,
+		nil,
+		wrappers.OPEN_EXISTING,
+		openFlags,
+		0)
+	if e != nil {
+		return "", NewWindowsError("CreateFile", e)
+	}
+	defer wrappers.CloseHandle(file)
+
+	buf := make([]uint16, wrappers.MAX_PATH)
+	if _, err = wrappers.GetFinalPathNameByHandle(file, &buf[0], wrappers.MAX_PATH, finalPathFlags); err != nil {
+		return "", NewWindowsError("GetFinalPathNameByHandle", err)
+	}
+	result = syscall.UTF16ToString(buf)
+	return result, err
+}
+
+// GetFinalPathNameAsDosName returns symlik target in "DOS" format (c:\dir\name) or source fileName if fileName is normal,
+// not symlinked file
+func GetFinalPathNameAsDosName(fileName string) (string, error) {
```

Comment:
[Nitpick] Rename to `GetFinalPathNameAsDOSName` for consistency with existing names like `VerFileDOS`.